pkg/k8s/k8s_namespace.go

```diff
@@ -0,0 +1,66 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// InitializeNamespace sets up the namespace based on the KNUU_DEDICATED_NAMESPACE environment variable
+func InitializeNamespace(identifier string) (string, error) {
+	useDedicatedNamespace, err := strconv.ParseBool(os.Getenv("KNUU_DEDICATED_NAMESPACE"))
+	if err != nil {
+		useDedicatedNamespace = false
+	}
+
+	var namespaceName string
+	if useDedicatedNamespace {
```

Comment:
q: Do we need this check in this function as it's only called if the env var is set to `true`?